src/meshcat/servers/zmqserver.py

```diff
@@ -137,6 +192,42 @@ def f(port):
         self.web_url = "{protocol}//{host}:{port}/static/".format(
             protocol=protocol, host=self.host, port=self.fileserver_port)
 
+        # Note: The (significant) advantage of putting this in here is not only
+        # so that the workflow is convenient, but also so that the server
+        # administers the public web_url when clients ask for it.
+        if ngrok_http_tunnel:
+            if protocol == "https:":
+                # TODO(russt): Consider plumbing ngrok auth through here for
+                # someone who has paid for ngrok and wants to use https.
+                raise(Exception('The free version of ngrok does not support https'))
+
+            # Conditionally import pyngrok
+            try:
+                import pyngrok.conf
+                import pyngrok.ngrok
+
+                kwargs = {}
+                # Use start_new_session if it's available. Without it, in
+                # jupyter the server goes down when we cancel execution of any
+                # cell in the notebook.
+                if sys.version_info.major >= 3:
+                        kwargs['start_new_session'] = True
+                config = pyngrok.conf.PyngrokConfig(**kwargs)
+                self.web_url = pyngrok.ngrok.connect(self.fileserver_port, "http", pyngrok_config=config) + "/static/"
+                print("\n")  # ensure any pyngrok output is properly terminated.
+
+                def cleanup():
+                    pyngrok.ngrok.kill()
+
+                atexit.register(cleanup)
+
+            except ImportError as e:
+                if "pyngrok" in e.__class__.__name__:
+                    raise(Exception("You must install pyngrok (e.g. via `pip install pyngrok`)."))
+            except:
```

Comment:
The generic `except....raise` here is unnecessary. Python will already do the right thing (raise the exception) if it isn't captured by an existing `except` clause. For example:

```py
In [1]: x = {}

In [2]: try:
   ...:     x["a"]
   ...: except ValueError as e:
   ...:     print("nope")
   ...:     
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
```

The `KeyError` is still raised as you would expect.